### ISSUE TITLE
Disagreement between Agadir server and pyAGADIR

### ISSUE THREAD

**Issue by rcrehuet** [reisalex/pyAGADIR#4]
If I run the example sequence in the [Agadir server](http://agadir.crg.es) I get the following result:
```
Thanks for using AGADIR
http://agadir.crg.es
Copyright @1997-2002, Lacroix E., Munoz V., Petukhov M. & Serrano, L.

pH                   7
Temperature          278
Ionic Strength       0.1

Nterm                free
Cterm                free

Peptide 1	ILKSLEEFLKVTLRSTRQT
res,  aa,  Hel,  Ncap,  Ccap,Hstaple,Schellm, CaH,  13Ca, JaN
-0.194007
01,   I,   0.0,  0.59,  0.00,  0.00,  0.00, -0.19 ,  0.00,  7.28
02,   L,   0.6,  0.54,  0.00,  0.00,  0.00,  0.05 ,  0.01,  6.61
03,   K,   1.1,  0.52,  0.00,  0.00,  0.00,  0.01 , -0.19,  6.43
04,   S,   1.7, 31.94,  0.00,  0.00,  0.00,  0.01 , -0.59,  7.17
05,   L,  33.6,  0.55,  0.00,  0.00,  0.00, -0.08 ,  0.59,  6.05
06,   E,  34.1,  0.76,  0.01,  0.00,  0.00, -0.12 ,  0.82,  5.71
07,   E,  34.9,  0.20,  0.02,  0.00,  0.00, -0.14 ,  0.84,  5.67
08,   F,  35.0,  0.13,  0.11,  0.00,  0.00, -0.13 ,  1.09,  6.32
09,   L,  34.9,  0.07,  0.23,  0.00,  0.00, -0.17 ,  0.63,  6.06
10,   K,  33.1,  0.03,  1.81,  0.00,  0.00, -0.12 ,  0.80,  6.04
11,   V,  32.7,  0.00,  0.44,  0.00,  0.00, -0.18 ,  1.13,  6.64
12,   T,  32.5,  0.07,  0.20,  0.00,  0.00, -0.18 ,  1.14,  6.79
13,   L,  31.1,  0.01,  1.46,  0.00,  0.00, -0.12 ,  0.59,  6.88
14,   R,   7.7,  0.00, 23.38,  0.00,  0.00, -0.02 ,  0.93,  8.04
15,   S,   1.7,  0.00,  6.02,  0.00,  0.00, -0.00 ,  0.21,  6.98
16,   T,   1.6,  0.00,  0.10,  0.00,  0.00, -0.01 ,  0.06,  7.44
17,   R,   1.1,  0.00,  0.52,  0.00,  0.00, -0.00 ,  0.05,  6.51
18,   Q,   0.0,  0.00,  1.11,  0.00,  0.00,  0.04 ,  0.03,  6.46
19,   T,   0.0,  0.00,  0.00,  0.00,  0.00, -0.16 ,  0.00,  7.46
Percentage helix	18.68
```
whereas using the `r` model gives these helicities:
`0.9 2.1 4.1 14.5 16.5 17.5 17.9 17.9 18.0 17.6 14.8 12.9 11.3 8.6 2.9 1.7 1.2 0.6 0.1`
And the `1s` models these ones:
`0.7 1.7 3.5 13.8 16.1 17.4 17.8 17.9 17.9 17.5 14.3 12.2 10.4 7.7 2.5 1.4 1.0 0.5 0.1`
which are about half the values of the Agadir server. @mengqvist mentioned in this issue  https://github.com/reisalex/pyAGADIR/issues/2  that he was getting values half the experimental ones. Could this be related?
The problem is the server does not specify which values it is using, and it cites all Agadir publications, which really does not help.
Do you know if the disagreement is only caused by the use of different models (surprising for such a large difference) or could there be an error in the code?

**Comment by mengqvist** [comment]
Oooh! This data is gold! I can use the individual scores to tweak the code further. I was missing something like this. Perhaps you could run a couple more peptides, if I need more data down the line?

I'll try to have a look at it this week.

In terms of accuracy of AGADIR I feel like I did some significant improvements to the code in spring. All of that is still stuck in a pull request though (https://github.com/reisalex/pyAGADIR/pull/3).

**Comment by mengqvist** [comment]
To answer your questions about errors in the code I would say yes, there are some. But that's mainly because it's really hard to parse the meaning of all those different publications. Been trying to improve on that to get correct predictions.

**Comment by mengqvist** [comment]
Maybe those values are not as useful as I thought. My expectation was that these would be energy terms. But I can't get it to make sense. It seems to me that the are not energy terms, but rather helicity values on a per residue basis. 

The values I get in my revised version PR #3 are closer to what you get from the webserver, but still not correct.

res aa helix
1 I 0.00
2 L 0.92
3 K 1.69
4 S 3.16
5 L 29.31
6 E 29.86
7 E 30.39
8 F 30.69
9 L 30.42
10 K 29.36
11 V 24.27
12 T 19.34
13 L 17.60
14 R 11.32
15 S 3.89
16 T 2.25
17 R 1.37
18 Q 0.67
19 T 0.00

**Comment by rcrehuet** [comment]
Hi Martin. Do you mean to run them in the Agadir server? 

----- On 21 Aug, 2024, at 13:57, Martin Engqvist ***@***.***> wrote: 

| @rcrehuet If you would be so kind as to run the peptides below in 1s mode and
| share the result I might be able to debug the code.

| AAAAAAAA
| SAAAAAAA
| AAAAAAAG
| SAAAAAAG

| AAADAAAA
| AAAAKAAA
| AAAKKAAA
| AAADKAAA

| ADAAAKAA
| ADAAKAAA
| AIAAALAA
| AIAALAAA

| There are more, but these would be a good start.

| —
| Reply to this email directly, view it on GitHub , or unsubscribe .
| You are receiving this because you were mentioned. Message ID:
| <reisalex/pyAGADIR/issues/4/2301872453 @ github . com>

**Comment by mengqvist** [comment]
Yes, exactly! I need to know the correct output of some simple well-defined peptides. That way I can hopefully debug the code. Problem is I don't have an academic affiliation, so I can't run them myself.

**Comment by rcrehuet** [comment]
Hi Martin, 
I attach the results for each of the peptides. 
A problem with Agadir server is that it does not specify which model it is using, nor can you chose the model. So it could be that it is not using the 1s model. But hope this helps anyway. 
Cheers, 
Ramon 

----- On 2 Sep, 2024, at 12:17, Ramon Crehuet ***@***.***> wrote: 

| ----- On 26 Aug, 2024, at 08:40, Martin Engqvist ***@***.***>
| wrote:

|| Yes, exactly! I need to know the correct output of some simple well-defined
|| peptides. That way I can hopefully debug the code. Problem is I don't have an
|| academic affiliation, so I can't run them myself.

|| —
|| Reply to this email directly, view it on GitHub , or unsubscribe .
|| You are receiving this because you were mentioned. Message ID:
|| <reisalex/pyAGADIR/issues/4/2309445519 @ github . com>

**Comment by rcrehuet** [comment]
I think the files were not attached in the email reply, so for the sake of completeness, here they are.
[peptides.zip](https://github.com/user-attachments/files/16836250/peptides.zip)

**Comment by mengqvist** [comment]
Thanks Ramon! I'll get to work on this and see what I can do. 

I assume they are using the multiple sequence setting since that's their most recent one. For short peptides the result between 1s and ms are very similar according to their paper. (Figure 4 in https://doi.org/10.1002/(SICI)1097-0282(19970415)41:5%3C495::AID-BIP2%3E3.0.CO;2-H).

**Comment by reisalex** [comment]
Reviewing Martin's work now!

**Comment by reisalex** [comment]
Hi @rcrehuet we're investigating now. If you have any insights, please let us know.

As you note, the AGADIR webserver is not clear either as to which model it is using (presumably the newer model(s)). In this initial implementation I took a stab at the v1 model (from the 1994 paper). I had noticed that there wasn't perfect agreement, however the relative helicities seemed to be about right.

**Comment by rcrehuet** [comment]
Hi Alex. Thanks for your message. I am currently working with @mengqvist to improve the model and correct it.
